src/tests/ftest/server/multiengine_persocket.py

```diff
@@ -135,33 +134,112 @@ def daos_server_storage_prepare_ns(self, step, engines_per_socket=1):
         cmd.sudo = False
         cmd.debug.value = False
         cmd.set_sub_command("scm")
-        cmd.sub_command_class.set_sub_command("reset")
+        cmd.sub_command_class.set_sub_command("prepare")
         cmd.sub_command_class.sub_command_class.scm_ns_per_socket.value = engines_per_socket
         cmd.sub_command_class.sub_command_class.force.value = True
-        cmd.sub_command_class.sub_command_class.scm_only.value = True
 
         self.log.info(
-            "===(%s.B)Starting daos_server storage prepare -S: %s", step, str(cmd))
-        results = run_pcmd(self.hostlist_servers, str(cmd), timeout=180)
-        if results[0]['exit_status']:
+            "===(%s.B)Starting daos_server scm prepare -S: %s", step, str(cmd))
+        results = run_remote(self.log, self.hostlist_servers, str(cmd), timeout=180)
+        if not results.passed:
             self.fail(
                 "#({0}.B){1} failed, "
                 "please make sure the server equipped with {2} PMem "
                 "modules.".format(step, cmd, engines_per_socket))
-        time.sleep(15)
-        self.host_reboot(self.hostlist_servers)
+
+    def wait_for_result(self, get_method, timeout, **kwargs):
+        """Wait for a result with a timeout.
+
+        Args:
+            get_method (object): method to call to determine if the result is found
+            timeout (int): number of seconds to wait for a response to be found
+            kwargs (dict): kwargs for get_method.
+
+        Returns:
+            bool: if the result was found
+        """
+        result_found = False
+        timed_out = False
+        start = time.time()
+
+        while not result_found and not timed_out:
+            timed_out = (time.time() - start) >= timeout
+            result_found = get_method(**kwargs)
+            self.log.debug("==Response found by %s(%s): %s", get_method, kwargs, result_found)
+            time.sleep(5)
+        return not timed_out
+
+    def check_ping(self, host, expected_pings=0, cmd_timeout=60, verbose=True):
```

Comment:
My general recommendation would be to move wait_for_result, check_ping and check_ssh to a utils file. In this way, it can be utilized by other test cases.

Reply:
Please see the update.